plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -571,7 +572,7 @@ func (c *CloudWatch) BuildMetricDatum(point telegraf.Metric) []*cloudwatch.Metri
 // 30 dimensions per metric so we only keep up to the first 30 alphabetically.
 // This always includes the "host" tag if it exists.
 // See https://github.com/aws/amazon-cloudwatch-agent/issues/398
-func BuildDimensions(mTags map[string]string) []*cloudwatch.Dimension {
+func BuildDimensions(mTags map[string]string, globalDimensions map[string]string) []*cloudwatch.Dimension {
```

Comment:
One quick question @SaxyPandaBear: do you have opinions about the priority of the global tags vs. the `host` tag that's passed in via the per-metric dimension map (I'm thinking in terms of the max dimensions permitted here)? At the moment, the `host` has highest priority, then the global tags, then the other per-metric tags.

Reply:
I'll have to trace through it, considering I didn't know that `host` had a higher priority than the other metrics. I assumed it was equal priority before. I'll take a look today.

Reply:
While you're looking at this, I think the `sort.Strings(keys)` business is there to keep the dimensions used consistent when the number exceeds the max permitted (map iteration order is deliberately randomised in golang). I should probably add this to the global dimensions for the same reason.

Reply:
(plus there should probably be tests for that behaviour as well — these are currently missing for the existing code)

Reply:
The fact that the host tag is explicitly called out with this comment:
`// This is pretty ugly but we always want to include the "host" tag if it exists.` 

Makes me think this is something we need to preserve. For better or for worse. There's too much risk in changing this specific behavior. 

With respect to the key sorting, agreed. I think we'll need to sort the global dimensions to maintain consistent behavior. Good call out. I didn't catch that in the initial pass through

Reply:
Will wait to resolve this comment until the `sort.Strings` is added for the global dimension keys

Reply:
We can divide into two append_dimensions here:
* `append_dimensions` is for [each metric plugins](https://github.com/aws/amazon-cloudwatch-agent/blob/c85501042b088014ec40b636a8b6b2ccc9739738/translator/translate/metrics/util/commonconfigutil.go#L54-L57) which is added as a telegraf tag; therefore, will be added firstly as a tag
* `append_dimensions` which is the [ec2tagger](https://github.com/aws/private-amazon-cloudwatch-agent-staging/tree/b7f7db3cd86dd8e60c2911561b2d0f5622e86be0/plugins/processors/ec2tagger). Since its a processors, its will be added secondly as a tag
* Lastly would be your global dimensions which will added in the global dimensions

However, the [global dimensions are processed first](https://github.com/aws/amazon-cloudwatch-agent/blob/13af598623255e1dfebd5088212acf863c6b7476/plugins/outputs/cloudwatch/cloudwatch.go#L586-L596) so global dimensions will be prioritized before others `append_dimensions`

Reply:
I'm not sure I understand this comment @khanhntd — are you suggesting we make a change here or describing how the current implementation works?